src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpChannelFactory.cs

```diff
@@ -556,7 +558,7 @@ private void InitializeSecurityTokenManager()
 
         protected virtual bool IsSecurityTokenManagerRequired()
         {
-            return AuthenticationScheme != AuthenticationSchemes.Anonymous;
+            return ProxyAuthenticationScheme != AuthenticationSchemes.Anonymous || AuthenticationScheme != AuthenticationSchemes.Anonymous;
```

Comment:
From looking at the .NET Framework code, the correct method code here looks to be:
```c#
            if (AuthenticationScheme != AuthenticationSchemes.Anonymous)
            {
                return true;
            }
            if (_proxyFactory != null && _proxyFactory.AuthenticationScheme != AuthenticationSchemes.Anonymous)
            {
                return true;
            }
            else
            {
                return false;
            }
```

Reply:
OK, I'll check that with my infrastructure setup and adapt the changes, thanks

Reply:
Solved in https://github.com/dotnet/wcf/pull/4555/commits/a10eff559b2cac90214e15d1c55c05b8428d1740

I can confirm that the change works in my enterprise's infrastructure